dali/operators/input/video_input.h

```diff
@@ -402,12 +405,15 @@ void VideoInput<Backend, FramesDecoder>::VideoInputRunImpl(Workspace &ws) {
        * "next_output_data_id" trace.
        */
       LoadDataFromInputOperator(GetThreadPool(ws));
+    } else {
+      will_return_next = false;
```

Comment:
`input_sample_depleted` tells, if the input encoded video has ended. `will_return_next` OTOH tells, that the operator can be run again, without the need of feeding it. The difference between those will appear when user feeds 2 videos beforehand. Then, after the first one ends, we'll have a state:
```c++
input_sample_depleted = true   // because the video ended
SetDepletedOperatorTrace(false)   // because there is another video in queue
```